addons/website_sale_collect/models/sale_order.py

```diff
@@ -42,7 +51,7 @@ def _check_cart_is_ready_to_be_paid(self):
         if (
             self._has_deliverable_products()
             and self.carrier_id.delivery_type == 'in_store'
-            and not self._is_in_stock(self.pickup_location_data['warehouse_id'])
+            and not self._is_in_stock(self.pickup_location_data['id'])
```

Comment:
Why don't use `self.warehouse_id` now that you set it when selecting the location?
```suggestion
            and not self._is_in_stock(self.warehouse_id)
```